sunbird-es-utils/src/main/java/org/sunbird/common/ElasticSearchRestHighImpl.java

```diff
@@ -0,0 +1,907 @@
+   * @param data Map<String,Object>
+   * @return true or false
+   */
+  @Override
+  public Future<Boolean> update(
+      String index, String type, String identifier, Map<String, Object> data) {
+    long startTime = System.currentTimeMillis();
+    ProjectLogger.log(
+        "ElasticSearchRestHighImpl:update method started at ==" + startTime + " for Type " + type,
+        LoggerEnum.PERF_LOG);
+    Promise<Boolean> promise = Futures.promise();
+    ;
+
+    if (!StringUtils.isBlank(index)
+        && !StringUtils.isBlank(type)
+        && !StringUtils.isBlank(identifier)
+        && data != null) {
+      Map<String, String> mappedIndexAndType =
+          ElasticSearchHelper.getMappedIndexAndType(index, type);
+      UpdateRequest updateRequest =
+          new UpdateRequest(
+                  mappedIndexAndType.get(JsonKey.INDEX),
+                  mappedIndexAndType.get(JsonKey.TYPE),
+                  identifier)
+              .doc(data);
+
+      ActionListener<UpdateResponse> listener =
+          new ActionListener<UpdateResponse>() {
+            @Override
+            public void onResponse(UpdateResponse updateResponse) {
+              promise.success(true);
+              if (updateResponse.getResult() == DocWriteResponse.Result.CREATED
+                  || updateResponse.getResult() == DocWriteResponse.Result.UPDATED
+                  || updateResponse.getResult() == DocWriteResponse.Result.NOOP) {
+                ProjectLogger.log(
+                    "ElasticSearchRestHighImpl:update  Success with upsert for index : "
+                        + index
+                        + ",type : "
+                        + type
+                        + ",identifier : "
+                        + identifier,
+                    LoggerEnum.INFO);
+              }
+              long stopTime = System.currentTimeMillis();
+              long elapsedTime = stopTime - startTime;
+              ProjectLogger.log(
+                  "ElasticSearchRestHighImpl:update method end at =="
+                      + stopTime
+                      + " for Type "
+                      + type
+                      + " ,Total time elapsed = "
+                      + elapsedTime,
+                  LoggerEnum.PERF_LOG);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+              ProjectLogger.log(
+                  "ElasticSearchRestHighImpl:update exception occured:" + e.getMessage(),
+                  LoggerEnum.ERROR.name());
+              promise.failure(e);
+            }
+          };
+      ConnectionManager.getRestClient().updateAsync(updateRequest, listener);
+
+    } else {
+      ProjectLogger.log(
+          "ElasticSearchRestHighImpl:update Requested data is invalid.", LoggerEnum.INFO.name());
+      promise.failure(
+          new ProjectCommonException(
+              "", "Requested data is invalid", ResponseCode.invalidData.getResponseCode()));
```

Comment:
use constant to define this: Requested data is invalid

Reply:
Done